promql/ast.go

```diff
@@ -231,6 +231,9 @@ type VectorMatching struct {
 	// On contains the labels which define equality of a pair
 	// of elements from the vectors.
 	On model.LabelNames
+	// Ignoring excludes the given label names from matching,
+	// rather than only using them.
+	Ignoring bool
```

Comment:
I was thinking maybe a different name for `On`, but otherwise I think a map and a bool is the way to go here from a cleanliness standpoint.